### PR TITLE
Add script to help test configgin changes locally

### PR DESCRIPTION
**Pull request by mook-as**

### CODE REVIEW

File: hack.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+
+# This will rebuild configgin for the image
+set -o errexit -o nounset
+
+REPO="${REPO:-scf}"
+
+IMAGE="$( cd "../${REPO}" && source .envrc && echo "$FISSILE_STEMCELL" )"
```
Comment (jandubois) [inline hack.sh]: `${FISSILE_STEMCELL}`

File: hack.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+export KUBECONFIG="$(mktemp)"
+kubectl config set-cluster vagrant --server=http://cf-dev.io:8080
+kubectl config set-context vagrant --cluster=vagrant --user=""
+kubectl config use-context vagrant
+
+vagrant_ready=""
+if test -z "${NO_RUN:-}" ; then
+    if ( cd "$(dirname "$0")/../${REPO}" && (vagrant status 2>/dev/null | grep --quiet running) ) ; then
+        vagrant_ready="true"
+        helm list --short | xargs --no-run-if-empty helm delete --purge
```
Comment (jandubois) [inline hack.sh]: Doesn't work on macOS. Workaround:

```
        releases=$(helm list --short)
        test -n "${releases}" && helm delete --purge ${releases}
```

File: hack.sh [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    --detach \
+    "${name}:${tag}-orig" \
+    /bin/bash -c "sleep 1d")
+docker exec -t "${container}" /bin/bash -c "source /usr/local/rvm/scripts/rvm && make -C /src all"
+docker exec -t "${container}" /bin/bash -c "source /usr/local/rvm/scripts/rvm && gem install /src/configgin-*.gem"
+docker commit "${container}" "${IMAGE}"
+
+test -z "${NO_RUN:-}" || exit
+
+docker_user=$(docker system info | awk -F: '{ if ($1 == "Username") { print  $2} }' | tr -d '[:space:]' ||:)
```
Comment (jandubois) [inline hack.sh]: Doesn't work on OS X. Maybe:

```
 test -z "${NO_RUN:-}" || exit
-
-docker_user=$(docker system info | awk -F: '{ if ($1 == "Username") { print  $2} }' | tr -d '[:space:]' ||:)
-if test -n "${docker_user}" ; then
-    docker tag "${IMAGE}" "${docker_user}/${name##*/}:${tag}"
-    docker push "${docker_user}/${name##*/}:${tag}"
+if [ "$(uname)" == "Darwin" ]; then
+    docker_user=$(docker-credential-osxkeychain list | jq -r '.["https://index.docker.io/v1/"]')
+else
+    docker_user=$(docker system info | awk -F: '{ if ($1 == "Username") { print  $2} }' | tr -d '[:space:]' ||:)
 fi
+if test -z "${docker_user}" ; then
+    echo "Can't determine docker user. Are you logged in?"
+    exit 1
+fi
+docker tag "${IMAGE}" "${docker_user}/${name##*/}:${tag}"
+docker push "${docker_user}/${name##*/}:${tag}"

 test -n "${vagrant_ready}" || exit
```

File: hack.sh [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+if test -n "${docker_user}" ; then
```
Comment (jandubois) [inline hack.sh]: The `vagrant ssh` code later relies on the username being set:

```
docker pull ${docker_user}/${name##*/}:${tag}
```

Therefore the patch above aborts when the user is not logged in.

File: hack.sh [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    docker tag "${IMAGE}" "${docker_user}/${name##*/}:${tag}"
+    docker push "${docker_user}/${name##*/}:${tag}"
+fi
+
+test -n "${vagrant_ready}" || exit
```
Comment (jandubois) [inline hack.sh]: Why is this needed; there is a check if vagrant is running just 3 lines later...

Comment (mook-as) [reply hack.sh]: That was for the `NO_RUN` checking. I'll drop the check 3 lines later.

**Pull request merged**